scala/BUILD

```diff
@@ -0,0 +1,20 @@
+load("//scala:scala_toolchain.bzl", "scala_toolchain")
+
+#call scala toolchain
+scala_toolchain(
+      name = 'scala_toolchain_impl',
```

Comment:
I think this is a collision too. I think we can name this scala_default_toolchain_impl or something to not confused with the toolchain rule impl.